### PR TITLE
feat: matka.fi separate park and ride options for car and bike

### PR DESCRIPTION
**Pull request by tekoiv**

### CODE REVIEW

File: app/configurations/config.matka.js [hunk 1]
```diff
@@ -288,7 +288,9 @@ export default {
 
   includeCarSuggestions: true,
   includeParkAndRideSuggestions: true,
-
+  // DT-5469
+  includeBikeParkSuggestions: true,
```
Comment (vesameskanen) [inline app/configurations/config.matka.js]: one switch is enough. Use existing showBikeAndParkItineraries.

**Review by vesameskanen — CHANGES_REQUESTED** [review]
- Car parkride switch has disappeared from HSL planner
- Matka planner requests bikeparkride itinerariers even when the respective switch is turned off. This is a computation intensive feature so bikeparkplan should be included only when user asks them.

**Review by vesameskanen — APPROVED** [review]

File: app/component/customizesearch/StreetModeSelectorPanel.js [hunk 2]
```diff
@@ -62,43 +66,75 @@ const StreetModeSelectorPanel = (
...
                     }
                   />
-                </label>
+                </span>
+              </label>
+            </div>
+          ) : null}
+        </div>
+        {config.includeParkAndRideSuggestions &&
+          !config.separatedParkAndRideSwitch && (
```
Comment (derhuerst) [inline app/component/customizesearch/StreetModeSelectorPanel.js:109]: I'm not sure I understand the logic, but shouldn't this boolean flag work in the exact opposite way? As in: If the "separated switch" config is `true`, it should show the switch in the UI?
We stumbled upon this while merging changes into the [bbnavi code](https://github.com/bbnavi/digitransit-ui).

Comment (vesameskanen) [reply app/component/customizesearch/StreetModeSelectorPanel.js:109]: In reittiopas.hsl.fi, car symbol means park and ride. There is no plain car routing available.
in other planners ( say tampere.digitransit.fi ), car symbol means car routing and there is additional switch for park & ride.

Configuration for e.g. tampere says: 
  // Park and ride and car suggestions separated into two switches                                                                                                                                         
  separatedParkAndRideSwitch: true,

So the switch naming logic seems to be right. The rendering code might be strange as and identical element is used for two different purposes.

Such inconsistency is unfortunate but comes from customer's design.

Comment (derhuerst) [reply app/component/customizesearch/StreetModeSelectorPanel.js:109]: Oh, we missed the part below, thanks for pointing this out. In any case, I'd strongly advocate for DRYing the two (quite similar but subtly different) code sections.

**Pull request merged**